taigit/src/libraries/GitHub/src/cyclomaticComplexity.ts

```diff
@@ -0,0 +1,141 @@
+/*
+Calculates McCabe cyclomatic complexity of a repository
+get cyclomatic complexity of python code
+*/
+
+import axios from 'axios';
+
+/**
+ *  check if the file is in the list of approved formats
+ * @param fileExt file extension of the specific java file
+ * @return bool true if the file is a java, python or c++ file
+ */
+function 
+verifyFormat(fileExt: string) : boolean{
+   let bool : boolean = false; 
+   if (fileExt === "java" || fileExt === "py" || fileExt === "cpp"){
+       bool = true;
+   }
+   return bool;
+}
+
+/**
+ * 
+ * calculates code  
+ * @param file 
+ * @see https://radon.readthedocs.io/en/latest/intro.html#cyclomatic-complexity
+ * @return num points in total
+ */
+
+function 
+ getCodeComplexity(code: string) : number{
+    let num : number = 0;
+    let ignoreComment = 0;
+    let arr : Array<string> = code.split(" ");
+    let arr1 : Array<string> = ["if", "elif", "case", "for", "while", "return", "void",
+                                "boolean", "bool", "assert", "except", "with", "iter"];
+    for(var i = 0; i < arr.length; i++){
+        if(ignoreComment == 0){
+            if(arr1.includes(arr[i])){
+                num++;
+            }else if(arr[i] === "else" && arr[i+2] === "if"){
+                num++;
+                i = i+2;
+            }else if(arr[i]==="/*" || arr[i] === "/**"){
+                ignoreComment = 1;
+            }
+        }else if(arr[i] ==="*/" || arr[i] === "**/"){
+            ignoreComment = 0;
+        }  
+    }
+    return num;
+ }
+
+
+/**
+ * Creates a json file that contains the cyclomatic complexity of a file
```

Comment:
I would elaborate on this. I see sites discuss CC of functions, but typically not files. be very specific to front end about what they're recieving, and make sure they know that it's a beta feature.